tests/unit/jobserver/test_icons.py

```diff
@@ -2,29 +2,48 @@
 import re
 from pathlib import Path
 
+import pytest
 
-def test_all_icons(subtests):
-    """Ensure icon tag names defined in templates are valid tags"""
+
+# Logically, this feels like it could be a pytest.fixture.
+# However, it's not currently possible to use fixtures with parametrize:
+# https://github.com/pytest-dev/pytest/issues/349
+def icons():
     templates = [p for p in Path("templates").glob("**/*.html")]
 
-    # remove templates with no icon tag so we can simplify the regex matching
-    # code below
+    # Remove templates with no icon tag so we can simplify the regex matching
+    # code below.
     icon_templates = [p for p in templates if "{% icon_" in p.read_text()]
 
-    # look for the start of an icon templatetag and capture the full name
+    # Look for the start of an icon templatetag and capture the full name.
     icon_pat = re.compile(r"{% (icon_[^\s]+)")
 
-    # get a set of icon names in the Django templates
-    icons = set(
-        itertools.chain.from_iterable(
-            icon_pat.search(t.read_text()).groups() for t in icon_templates
+    # Get a sorted list of non-duplicated icon names in the Django templates.
+    # We use a sorted list instead of only a set
+    # to make this test work with parametrize and xdist:
```

Comment:
We can be more precise here and help the reader understand the point:
```suggestion
    # Use a sorted list so that the different xdist workers get a consistent ordering.
```